File: src/java/nz/co/searchwellington/spam/SpamFilter.java
package nz.co.searchwellington.spam;

import nz.co.searchwellington.model.Resource;

public class SpamFilter {
    
    public boolean isSpam(Resource editResource) {
        if (editResource != null) {
            boolean urlIsSpam = editResource.getUrl() != null && editResource.getUrl().contains("rfid");
            boolean descriptionsSpam = editResource.getDescription() != null && editResource.getDescription().contains("rfid");
            return urlIsSpam || descriptionsSpam;
        }
        return false;
    }
    
}

File: src/main/scala/nz/co/searchwellington/spam/SpamFilter.scala
package nz.co.searchwellington.spam

import org.springframework.stereotype.Component
import nz.co.searchwellington.model.Resource

@Component
class SpamFilter {
  def isSpam(editResource: Resource): Boolean = {
    val urlIsSpam = editResource.page.contains("rfid")
    val descriptionsSpam = editResource.description.exists(d => d.contains("rfid"))
    urlIsSpam || descriptionsSpam
  }
}
